flow/Arena.h

```diff
@@ -510,9 +510,15 @@ class StringRef {
 	int expectedSize() const { return size(); }
 
 	int compare(StringRef const& other) const {
-		size_t minSize = std::min(size(), other.size());
-		if (minSize != 0) {
-			int c = memcmp(begin(), other.begin(), minSize);
+		return compareSuffix(other, 0);
```

Comment:
Given how often this function is used in general in FDB, we should make sure this is not worse than before.  Ideally it would be fully inlined and the `- prefixLen` would be optimized out since prefixLen would be a constant 0.  I'm not sure what would happen with the assert, or why it's written the way it is - perhaps to avoid being optimized out?  It is expanded to a loop with an if / throw inside it.

It might be easier to just leave it alone and have two methods which call memcmp slightly differently, with slightly different precondition checks.

Reply:
Yeah, I actually originally wrote compareSuffix, and decided to refactor it after, assuming that the extra function call and the subtraction would get inlined/optimized. I'm happy to change it back.
I can also remove the assert.